packages/back-end/src/services/informationSchema.ts

```diff
@@ -88,21 +89,21 @@ export async function mergeStaleInformationSchemaWithUpdate(
     database.schemas.forEach((schema) => {
       const correspondingSchemaIndex = staleInformationSchema[
         correspondingIndex
-      ].schemas.findIndex(
+      ]?.schemas.findIndex(
```

Comment:
I'm wondering if in addition to adding these optional chaining operators, we update the early return logic so it also early returns if correspondingIndex === -1.

It shouldn't change any behavior, but I think it could make for a better dev experience since it'd be more clear that if there isn't a correspondingIndex, we don't need to look any further.

E.G. Update the early return on line 88 from:

```ts
if (!database.schemas) return;
```
to 

```ts
if (!database.schemas || correspondingIndex === -1) return;
```

Reply:
Yeah, let's do both.